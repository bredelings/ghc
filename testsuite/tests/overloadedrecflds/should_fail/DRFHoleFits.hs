{-# LANGUAGE DuplicateRecordFields #-}

module DRFHoleFits where

data T = MkT { foo :: Int }

bar = _ :: T -> Int
