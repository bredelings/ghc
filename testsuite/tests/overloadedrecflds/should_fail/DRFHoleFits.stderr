
DRFHoleFits.hs:7:7: error:
    • Found hole: _ :: T -> Int
    • In the expression: _ :: T -> Int
      In an equation for ‘bar’: bar = _ :: T -> Int
    • Relevant bindings include
        bar :: T -> Int (bound at DRFHoleFits.hs:7:1)
      Valid hole fits include
        foo :: T -> Int (defined at DRFHoleFits.hs:5:16)
        bar :: T -> Int (defined at DRFHoleFits.hs:7:1)
